language_generators/scala-defref-extractor/src/test/scala/io/bazeltools/buildgen/javadefref/JavaSourceEntityExtractorTest.scala

```diff
@@ -0,0 +1,114 @@
+package io.bazeltools.buildgen.javadefref
+
+import io.bazeltools.buildgen.shared.{DataBlock, Entity}
+import scala.collection.immutable.SortedSet
+
+import cats.effect.IO
+import io.circe.parser.decode
+
+class JavaSourceEntityExtractorTest extends munit.CatsEffectSuite {
+
+  case class DefsRefs(defs: SortedSet[Entity], refs: SortedSet[Entity])
+
+  def extractString(in: String): IO[DataBlock] =
+    JavaSourceEntityExtractor.extract(in)
+
+  def ents(s: String): SortedSet[Entity] =
+    decode[List[Entity]](s).map(_.to(SortedSet)) match {
+      case Right(ss) => ss
+      case Left(err) => sys.error(err.toString)
+    }
+
+  def struct(code: String): IO[DefsRefs] =
+    extractString(code).map { e => DefsRefs(e.defs, e.refs) }
+
+  test("test basic Java Example") {
+    val got = struct("""
+    package foo.bar;
+
+    import some.Pack;
+
+    class Bar {
+      static class Quux extends Pop {}
+      void run(Baz b) {
+        call(Bippy.foo);
+      }
+    }
+    """)
+
+    val expected = DefsRefs(
+        defs = ents("""["foo.bar.Bar"]"""),
+        // TODO: if we call b.call inside a method we shouldn't see b as a dep
+        //refs = ents("""["b", "foo.bar.b", "foo.bar.Baz", "Baz", "foo.bar.Pop", "Pop", "Bippy", "foo.bar.Bippy", "some.Pack"]"""))
+        refs = ents("""["foo.bar.Baz", "Baz", "foo.bar.Pop", "Pop", "Bippy", "foo.bar.Bippy", "some.Pack"]"""))
+
+    got.assertEquals(expected)
+  }
+
+  /*
```

Comment:
this test currently fails. It seems unref is not working in the new code, so we may not be able to remove phantom references.

Reply:
I mentioned this in slack, but the semantic knowledge of what ref/unref and so on doesn't exist in the extractors at all anymore. I didn't include unref i don't think since we haven't been using it. But that handling occurs downstream

Reply:
okay, I fixed the test and will merge when it goes green.